proguard6.0.3/core/src/proguard/configuration/ConfigurationLoggingInstructionSequenceConstants.java

```diff
@@ -0,0 +1,479 @@
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+package proguard.configuration;
+
+import proguard.classfile.*;
+import proguard.classfile.constant.Constant;
+import proguard.classfile.editor.*;
+import proguard.classfile.instruction.Instruction;
+import proguard.classfile.util.*;
+import proguard.optimize.peephole.InstructionSequenceReplacer;
+
+import static proguard.optimize.peephole.InstructionSequenceReplacer.catch_;
+
+/**
+ * This class contains a set of instruction sequences for accessing class information via reflection, and replacement instructions that add logging information on the reflection that is used.
+ *
+ * @author Johan Leys
+ */
+public class ConfigurationLoggingInstructionSequenceConstants
+{
+    private static String LOGGER_CLASS_NAME = ClassUtil.internalClassName(ConfigurationLogger.class.getName());
+
+    // Exceptions classes.
+    public static final String NAME_CLASS_NOT_FOUND_EXCEPTION     = "java/lang/ClassNotFoundException";
+    public static final String NAME_NO_SUCH_FIELD_EXCEPTION       = "java/lang/NoSuchFieldException";
+    public static final String NAME_NO_SUCH_METHOD_EXCEPTION      = "java/lang/NoSuchMethodException";
+    public static final String NAME_RUNTIME_EXCEPTION             = "java/lang/RuntimeException";
+    public static final String NAME_UNSATISFIED_LINK_ERROR        = "java/lang/UnsatisfiedLinkError";
+    public static final String NAME_IO_EXCEPTION                  = "java/io/IOException";
+
+    // Matched constants.
+    public static final int CLASS_NAME             = 0x30000000;
+    public static final int LOCAL_VARIABLE_INDEX_1 = 0x30000001;
+    public static final int LOCAL_VARIABLE_INDEX_2 = 0x30000002;
+    public static final int LOCAL_VARIABLE_INDEX_3 = 0x30000003;
+
+    public static final int CONSTANT_INDEX = InstructionSequenceMatcher.X;
+    public static final int ACCESS_MODE    = InstructionSequenceMatcher.Y;
+
+    public final Instruction[][][] RESOURCE;
+    public final Constant[]        CONSTANTS;
+
+    // Labels.
+    private final InstructionSequenceReplacer.Label TRY_START = InstructionSequenceReplacer.label();
+    private final InstructionSequenceReplacer.Label TRY_END   = InstructionSequenceReplacer.label();
+    private final InstructionSequenceReplacer.Label CATCH_END = InstructionSequenceReplacer.label();
+
+    private final InstructionSequenceReplacer.Label CLASS_NOT_FOUND_EXCEPTION;
+    private final InstructionSequenceReplacer.Label NO_SUCH_METHOD_EXCEPTION;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'NO_SUCH_METHOD_EXCEPTION' could be replaced by a local variable.](https://app.codacy.com/manual/airsquared/blobsaver/pullRequest?prid=5513820)